spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkPositionDeltaWrite.java

```diff
@@ -326,12 +328,7 @@ private void commitOperation(SnapshotUpdate<?> operation, String description) {
 
       CommitMetadata.commitProperties().forEach(operation::set);
 
-      if (wapEnabled && wapId != null) {
-        // write-audit-publish is enabled for this table and job
-        // stage the changes without changing the current snapshot
-        operation.set(SnapshotSummary.STAGED_WAP_ID_PROP, wapId);
-        operation.stageOnly();
```

Comment:
I mentioned this above, but i'm not comfortable extracting `stageOnly` here. That seems a bit different than "setWapProperties". Perhaps a rename may make this a bit more clear, but currently it feels like we are hiding an important state change on `operation`.